configs/raspberrypi/uefi/files/mender-rpi-grub-integration/usr/share/mender-rpi-grub-integration/init-runner

```diff
@@ -19,10 +19,21 @@
 
 reboot() {
     # RPi scripts call `reboot` with the `-f` argument, because it ordinarily expects to be run
-    # without systemd present. But in this setup it is present, and we need to do a proper
-    # non-forced reboot so that it shuts down correctly, including running `update-grub`, if
-    # necessary, before the reboot.
-    systemctl reboot
+    # without systemd present. But when using this init-runner, we are running under systemd. We
+    # could reboot, but that does not play nicely during upgrades, because the bootloader expects
+    # exactly one reboot only, so that might end up rolling back. However, at this point we should
+    # still be very early in the boot, so just switch over to the normal environment instead, which
+    # should be almost like a reboot. At the time of writing, the Raspberry Pi OS's `firstboot`
+    # script does a change to the partition UUID, which might not get picked up this way. However,
+    # Mender does not use this, so it should be fine to ignore.
+    #
+    # However, one thing we should do is to run `update-grub` to make sure that the changes that the
+    # `firstboot` script does to `cmdline.txt` are picked up. Let's do that using the dedicated
+    # service (better logging).
+    systemctl restart mender-update-grub-at-reboot
+
+    # Now switch to an ordinary boot.
+    systemctl default --no-block
```

Comment:
Consider squashing this last commit with the one that introduced the `reboot()` command in the first place.